test/ApiCheck.Test/BaselineComparerTests.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using ApiCheck.Baseline;
+using ApiCheckBaseline.V2;
+using Scenarios;
+using Xunit;
+
+namespace ApiCheck.Test
+{
+    public class BaselineComparerTests
+    {
+        public Assembly V1Assembly => typeof(ApiCheckBaselineV1).GetTypeInfo().Assembly;
+        public Assembly V2Assembly => typeof(ApiCheckBaselineV2).GetTypeInfo().Assembly;
+
+        public IEnumerable<Func<TypeInfo, bool>> TypeFilters => new Func<TypeInfo, bool>[]{
+            ti => !ti.Namespace.Equals("Scenarios")
+        };
+
+
+        [Fact]
+        public void Compare_Detects_ChangesInTypeVisibility()
+        {
+            // Arrange
+            var v1Baseline = CreateBaselineDocument(V1Assembly);
+            var v2Baseline = CreateBaselineDocument(V2Assembly);
+            var exceptions = CreateDefault();
+
+            var comparer = new BaselineComparer(v1Baseline, v2Baseline, BreakingChangeTypes.All, exceptions);
+
+            // Act
+            var changes = comparer.GetDifferences();
+
+            // Assert
+            var change = Assert.Single(changes, bc => bc.OldItem.Id == "public class ComparisonScenarios.PublicToInternalClass");
+        }
+
+        [Fact]
+        public void Compare_Detects_TypeRenames()
+        {
+            // Arrange
+            var v1Baseline = CreateBaselineDocument(V1Assembly);
+            var v2Baseline = CreateBaselineDocument(V2Assembly);
+            var exceptions = CreateDefault();
+
+            var comparer = new BaselineComparer(v1Baseline, v2Baseline, BreakingChangeTypes.All, exceptions);
+
+            // Act
+            var changes = comparer.GetDifferences();
+
+            // Assert
+            var change = Assert.Single(changes, bc => bc.OldItem.Id == "public interface ComparisonScenarios.TypeToRename");
+        }
+
+        [Fact]
+        public void Compare_Detects_TypeGenericityChanges()
+        {
+            // Arrange
+            var v1Baseline = CreateBaselineDocument(V1Assembly);
+            var v2Baseline = CreateBaselineDocument(V2Assembly);
+            var exceptions = CreateDefault();
+
+            var comparer = new BaselineComparer(v1Baseline, v2Baseline, BreakingChangeTypes.All, exceptions);
+
+            // Act
+            var changes = comparer.GetDifferences();
+
+            // Assert
+            var change = Assert.Single(changes, bc => bc.OldItem.Id == "public struct ComparisonScenarios.StructToMakeGeneric");
+        }
+
+        [Fact]
+        public void Compare_Detects_NamespaceChanges()
```

Comment:
While I do agree that a fundamental level that is what is happening, I want to make sure that the tests are worded in a higher level language that reflects the logical action that happened on the code base. (IE, I moved a type from namespace A into namespace B or nested it into another class and so on)